nion/swift/ExportDialog.py

```diff
@@ -246,66 +250,176 @@ def cancel(self) -> bool:
         return True
 
 
-class ExportSVGHandler:
-    def __init__(self, display_size: Geometry.IntSize) -> None:
-        self.width_model = Model.PropertyModel(display_size.width)
-        self.height_model = Model.PropertyModel(display_size.height)
+class UnitType(enum.Enum):
+    PIXELS = 0
+    INCHES = 1
+    CENTIMETERS = 2
+
+
+ConversionUnits = {
+    UnitType.PIXELS: 1,
+    UnitType.CENTIMETERS: 37,
+    UnitType.INCHES: 96
+}
 
-        self.int_converter = Converter.IntegerToStringConverter()
 
+class ExportSizeModel(Observable.Observable):
+    def __init__(self, display_item: DisplayItem.DisplayItem) -> None:
+        super().__init__()
+        display_size = self.__determine_display_size(display_item)
+        self.__width: int = display_size.width
+        self.__height: int = display_size.height
+        self.__aspect_ratio = self.__width / self.__height
+        self.__units = UnitType.PIXELS
+
+    def __determine_display_size(self, display_item: DisplayItem.DisplayItem) -> Geometry.IntSize:
+        if display_item.display_data_shape and len(display_item.display_data_shape) == 2:
+            return Geometry.IntSize(height=display_item.display_data_shape[0], width=display_item.display_data_shape[1])
+        return Geometry.IntSize(height=3, width=5)
```

Comment:
This is wrong. This function (which should be called `__calculate_display_size_in_pixels`) is returning a _pixel_size_ of 5x3 for line plots.